pkg/web/cookie/cookie.go

```diff
@@ -48,6 +49,8 @@ func (d *Cookie) new(r *http.Request) http.Cookie {
 	sameSite := d.SameSite
 	if sameSite == 0 {
 		sameSite = http.SameSiteLaxMode
+	} else if sameSite == http.SameSiteNoneMode && r.URL.Scheme == "http" {
```

Comment:
Isn't the front proxy using `http` to backend services? Won't that be seen as `http` here?

Reply:
This is correct. Proxies set `X-Forwarded-Proto` and `X-Forwarded-Host` headers which are extracted by the `ProxyHeaders` middleware. Our `RequestURL` middleware then fills the scheme and host in `r.URL`.

Reply:
Hm. So how to address this issue then?

Reply:
After giving this some more thought, I don't think we should "magically" downgrade security because a browser is (for whatever reason) using HTTP instead of HTTPS. If any other package tells `pkg/web/cookie` that it wants `SameSite=None`, we should respect that. So let's not make this change here.

Reply:
But what to do then for non TLS connections? Logging into the console is not possible currently due to this without TLS.

Reply:
At the moment, the only place where we use `SameSite=None` is the OAuth client, and I've already suggested a solution for that.